src/Common/src/TypeSystem/Common/TypeSystemException.cs

```diff
@@ -193,6 +193,10 @@ public static string OwningModule(MethodDesc method)
 
             public static string OwningModule(TypeDesc type)
             {
+#if TYPE_LOADER_IMPLEMENTATION
```

Comment:
The point is that NoMetadataType shouldn't even implement a Name and Namespace property because it will be a lie often times. `DefType` initially didn't even have Name and Namespace, but it made it very annoying because Canon types are just DefType, but we want to be able to get at their names like any other metadata based type.

Name and Namespace was not implemented on purpose. I'll talk to David.